### ISSUE TITLE
Breakpoint expression has no indication when it throws for invalid method call

### ISSUE THREAD

**Issue by FMorschel** [Dart-Code/Dart-Code#5196]
**Describe the bug**
Breakpoint expression has no indication when it throws for invalid method call.

**To Reproduce**

Code:

```dart
void main() {
  for (int i = 0; i < 10; i++) {
    print('Hello, World! $i');
    //i.isEvenn; // This is for uncommenting and validating that this is an error
  }
}
```

1. Create an empty project and add the above code.
2. Add a breakpoint in line 3 (`print`)
3. The expression for it should be the commented code on line 4: "`i.isEvenn`".
4. It will run the code and show no errors.

**Expected behavior**
Show some indication that the breakpoint expression is invalid.

**Screenshots**
![image](https://github.com/user-attachments/assets/6816aea7-019d-4253-86ef-faaff4f3ce42)

**Please complete the following information:**

<details>

<summary><strong>Workspace Environment</strong></summary>

```text
Dart Code extension: 3.93.20240711
Flutter extension: 3.93.20240702 (activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.91.1

Workspace type: Dart (LSP)
Workspace name: bug

Dart (3.6.0-36.0.dev): C:\Users\felip_0vh5fa6\.puro\shared\caches\850e0c1b16c3676c5b3da9bc7aa64fc165527c1d\dart-sdk
Flutter (undefined): C:\Users\felip_0vh5fa6\.puro\envs\master\flutter (No device)

```

</details>

<details>

<summary><strong>Output from 'dart info'</strong></summary>

`C:\Users\felip_0vh5fa6\.puro\shared\caches\850e0c1b16c3676c5b3da9bc7aa64fc165527c1d\dart-sdk\bin\dart.exe info`

If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.6.0-36.0.dev (dev) (Fri Jul 12 09:05:30 2024 -0700) on "windows_x64"
- on windows / "Windows 10 Pro" 10.0 (Build 22631)
- locale is pt-BR

#### Project info

- sdk constraint: '>=3.3.0 <4.0.0'
- dependencies: test
- dev_dependencies: dart_code_metrics, lints

#### Process info

| Memory | CPU | Elapsed time | Command line |
| -----: | --: | -----------: | ------------ |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |

</details>

**Comment by DanTup** [comment]
In a test with Dart, I actually do see errors here:

![image](https://github.com/user-attachments/assets/f951dee9-1138-44bd-a492-046c7568666a)

**Comment by DanTup** [comment]
And similar in Flutter (although the error is unfortunately less helpful):

![image](https://github.com/user-attachments/assets/38a24098-c413-435a-9961-80469757efe9)

**Comment by DanTup** [comment]
@FMorschel could you try capturing a log?

- Run the [**Dart: Capture Debugging Logs** command](https://dartcode.org/docs/logging/#capture-debugging-logs)
- Reproduce the issue
- Click **Cancel** on the logging notification to stop logging and open the log
- Remove anything sensitive from the log
- Attach the log here, in a [gist](https://gist.github.com/), or email to me at logs@dartcode.org.

Thanks!

**Comment by FMorschel** [comment]
Hi! I think I figured it out! When I took the screenshot above, I probably misclicked on "Run" and that would of course not show any debugging/breakpoint info. Even pressing `F5` would have been alright.

And I had previously seen that behaviour with a Flutter project but I was filtering out some unneeded lines from there and one of the keywords got that line!

If you wish to take a look at the Flutter result, we can keep this open, if not, we can close it. Thanks a lot!

**Comment by DanTup** [comment]
Yeah, let's keep this open to look more at the Flutter error messages. I suspect it's not specific to breakpoints and typing "x == 1" in the Debug Console would've produced the same result.

**Comment by FMorschel** [comment]
Hey! I have tested this on Flutter 3.24 and the message seems to be the same as Dart now. Not sure what changed. Will test this in an empty project JIC and then wait for your feedback to close this. (The reason I'll test in an empty project is that my project uses another isolate with no Flutter code, so _maybe_ something could be different but I doubt it.

**Comment by DanTup** [comment]
Hmmm, odd. Let me know what you see when you test and exactly what code/expression you're using.. Maybe the error was improved and I tested in mismatched versions of Dart/Flutter, but I can try to confirm.

**Comment by FMorschel** [comment]
The output is not exactly the _same_, but it is closer than before.

<details>

<summary> Flutter 3.24.0 - tested `i.isEvenn` on line 17 </summary>

Output:
```console
Debugger failed to evaluate breakpoint condition "i.isEvenn": error: org-dartlang-debug:synthetic_debug_expression:1:3: Error: The getter 'isEvenn' isn't defined for the class 'int'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'isEvenn'.
i.isEvenn
  ^^^^^^^
```

Code:

```
import 'package:flutter/material.dart';

void main() => runApp(const MaterialApp(home: Home()));

class Home extends StatefulWidget {
  const Home({super.key});

  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {
  int i = 0;

  @override
  Widget build(BuildContext context) {
    return const Scaffold();
  }
}
```

</details>

<details>

<summary> Dart 3.5.0 - tested `i.isEvenn` on line 3 </summary>

Output:

```console
Debugger failed to evaluate breakpoint condition "i.isEvenn": evaluateInFrame: (113) Expression compilation error
org-dartlang-debug:synthetic_debug_expression:1:3: Error: The getter 'isEvenn' isn't defined for the class '_Smi'.
 - '_Smi' is from 'dart:core'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'isEvenn'.
i.isEvenn
  ^^^^^^^
```

Code:

```dart
void main() {
  final i = 0;
  print(i);
}
```

</details>

**Comment by DanTup** [comment]
Yeah, I also cannot reproduce this. It feels similar to https://github.com/dart-lang/sdk/commit/ffbbdb5a10a73d598db54823bcc1ce3feda59d07 although those errors specifically said "collected" rather than "sentinel".

I suggest we close this while we have no repro and we can reinvestigate if it comes up again. There have been a lot of changes in VM I don't know all of the details between the previous and current stable, so it's possible something (deliberately or accidentally) improved this.
